engine/flink/management/periodic/src/test/scala/pl/touk/nussknacker/engine/management/periodic/UtilsSpec.scala

```diff
@@ -24,10 +30,38 @@ class UtilsSpec extends TestKit(ActorSystem("UtilsSpec")) with AnyWordSpecLike w
 
       val actorRef = system.actorOf(Props(new TestActor), actorName)
 
-      Utils.gracefulStopActor(actorRef, system)
+      Utils.stopActorAndWaitUntilItsNameIsFree(actorRef, system)
 
       // with normal system.stop(actorRef) or akka.pattern.gracefulStop this throws "actor name is not unique"
-      system.actorOf(Props(new TestActor), actorName)
+      val actorRef2 = system.actorOf(Props(new TestActor), actorName)
+
+      Utils.stopActorAndWaitUntilItsNameIsFree(actorRef2, system) // stop and free it so it doesn't impact other tests
+    }
+
+    "stop actor and free actor path without waiting for all of it's messages to be processed" in {
+      class TestActor extends Actor {
+        override def receive: Receive = { case msg =>
+          logger.info(s"Sleeping on the job '$msg' ...")
+          Thread.sleep(1000)
+        }
+      }
+      val actorName = "actorName"
+
+      val actorRef = system.actorOf(Props(new TestActor), actorName)
+
+      var messageCounter = 0
+      while (messageCounter < 1000) {
+        actorRef ! s"message number $messageCounter"
+        messageCounter += 1
+      }
+
+      Thread.sleep(1000)
+      // with gracefulStop this times out, because the PoisonPill is queued after the many normal messages
+      Utils.stopActorAndWaitUntilItsNameIsFree(actorRef, system)
+
+      val actorRef2 = system.actorOf(Props(new TestActor), actorName)
```

Comment:
Hmm this new test is failed.
![image](https://github.com/TouK/nussknacker/assets/534341/06f0899d-44e7-43e4-a5ad-959b2e835140)

Reply:
it only fails on CI, and I don't get why
are the tests on CI somehow "more parallel" than locally?

Reply:
I do get why, it's a race condition

i'll avoid it by just retrying actor creation instead of retrying actor stopping+freeing